test/unit/blur.js

```diff
@@ -91,4 +104,33 @@ describe('Blur', function () {
           });
       });
   });
+
+  it('invalid precision', function () {
+    assert.throws(function () {
+      sharp(fixtures.inputJpg).blur({ sigma: 1, precision: 'invalid' });
+    });
+  });
+
+  it('specific radius 10 and precision approximate', async () => {
+    const approximate = await sharp(fixtures.inputJpg)
+      .resize(320, 240)
+      .blur({ sigma: 10, precision: 'approximate' })
+      .toBuffer();
+    const integer = await await sharp(fixtures.inputJpg)
+      .resize(320, 240)
+      .blur(10)
+      .toBuffer();
+
+    assert.notDeepEqual(approximate, integer);
+
+    await new Promise(resolve => {
+      fixtures.assertSimilar(fixtures.expected('blur-10.jpg'), approximate, resolve);
```

Comment:
Could use `util.promisify()` here, although I should probably deal with this properly and modernise `assertSimilar` to return a Promise.

Reply:
Thought about refactoring `.assertSimilar` but it seemed out of scope of this PR, I probably can get to it in another PR if you'd like.

Reply:
No worries, I can fix that afterwards, no need to change anything in this PR.